File: src/main/scala/com/neighborhood/aka/laplace/estuary/bean/datasink/MysqlSinkBean.scala
package com.neighborhood.aka.laplace.estuary.bean.datasink

import com.neighborhood.aka.laplace.estuary.bean.credential.MysqlCredentialBean
import com.neighborhood.aka.laplace.estuary.core.sink.mysql.MysqlSinkFunc

/**
  * Created by john_liu on 2019/1/13.
  */
trait MysqlSinkBean extends DataSinkBean[MysqlSinkFunc] {

  def dataSinkType: String = SinkDataType.MYSQL.toString

  /**
    * 数据汇Mysql的链接信息
    */
  def credential: MysqlCredentialBean
}

File: src/main/scala/com/neighborhood/aka/laplace/estuary/web/bean/MysqlSinkBean.java
package com.neighborhood.aka.laplace.estuary.web.bean;



/**
 * Created by john_liu on 2019/1/17.
 */
public class MysqlSinkBean {

   private MysqlCredentialRequestBean credential;
    private boolean isAutoCommit;
    private long connectionTimeout;
    private int maximumPoolSize;

    public MysqlCredentialRequestBean getCredential() {
        return credential;
    }

    public void setCredential(MysqlCredentialRequestBean credential) {
        this.credential = credential;
    }

    public boolean isAutoCommit() {
        return isAutoCommit;
    }

    public void setAutoCommit(boolean autoCommit) {
        isAutoCommit = autoCommit;
    }

    public long getConnectionTimeout() {
        return connectionTimeout;
    }

    public void setConnectionTimeout(long connectionTimeout) {
        this.connectionTimeout = connectionTimeout;
    }

    public int getMaximumPoolSize() {
        return maximumPoolSize;
    }

    public void setMaximumPoolSize(int maximumPoolSize) {
        this.maximumPoolSize = maximumPoolSize;
    }
}
